### PR TITLE
Added hand_delivery flag to cases

### PR DESCRIPTION
**Pull request by ryangrundy7**
# Motivation and Context

Added hand delivery flag to the cases table. As we don't have all the treatment codes finalised, I've just put an example one of what it could be just to prove it works.

# What has changed

- Added hand delivery flag

# How to test?

- Run `mvn clean install`
- Run with other branches and run acceptance tests.

# Links

https://trello.com/c/774V6fGU/580-add-handdelivery-flag-to-casecreated-and-caseupdated-events-and-case-api-8
# Screenshots (if appropriate):

### CODE REVIEW

**Comment by ryangrundy7** [comment]
[add handDelivery flag to caseCreated and caseUpdated events and case API (8)](https://trello.com/c/774V6fGU/580-add-handdelivery-flag-to-casecreated-and-caseupdated-events-and-case-api-8)

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java [hunk 1]
```diff
@@ -114,4 +114,7 @@
   @Column(columnDefinition = "timestamp with time zone")
   @UpdateTimestamp
   private OffsetDateTime lastUpdated;
+
+  @Column(columnDefinition = "BOOLEAN")
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: you don't need the `columnDefinition` for the primative/common types

File: src/main/java/uk/gov/ons/census/casesvc/service/CaseService.java [hunk 2]
```diff
@@ -28,6 +19,12 @@
   private static final String HOUSEHOLD_INDIVIDUAL_RESPONSE_CASE_TYPE = "HI";
   public static final String CASE_UPDATE_ROUTING_KEY = "event.case.update";
 
+  // At the moment the treatment codes aren't finalised so there's an example one
+  // that tests it. Once the direct delivery treatment codes have been finalised then we can add
+  // them to the list.
+  private static final Set<String> DirectDeliveryTreatmentCodes =
+      new HashSet<>(Arrays.asList("CE_LDIEE"));
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/service/CaseService.java]: we could put these into `application.yml`. In fact, I think that's what we should do.

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java (hunk 1 above)
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: I notice `@Column(nullable = false, columnDefinition = "BOOLEAN DEFAULT false")` was added rather over-zealously in this entity file. It's only `Boolean` type which needs to be told it's not nullable and given a default. The primitive `boolean` is infinitely preferable for values which are non-nullable.

File: src/main/java/uk/gov/ons/census/casesvc/model/dto/CollectionCase.java [hunk 3]
```diff
@@ -33,4 +33,5 @@
   private Integer ceActualResponses;
   private Boolean addressInvalid;
   private Boolean undeliveredAsAddressed;
+  private Boolean handDelivery;
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/model/dto/CollectionCase.java]: make this `boolean` not `Boolean` to match the underlying entity

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java [hunk 4]
```diff
@@ -114,4 +114,6 @@
   @Column(columnDefinition = "timestamp with time zone")
   @UpdateTimestamp
   private OffsetDateTime lastUpdated;
+
+  @Column private boolean handDelivery;
```
Comment (AdamHawtin) [inline src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: Needs `@Column(nullable = false, columnDefinition = "BOOLEAN DEFAULT false")` to match the DDL

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java [hunk 5]
```diff
@@ -114,4 +114,7 @@
   @Column(columnDefinition = "timestamp with time zone")
   @UpdateTimestamp
   private OffsetDateTime lastUpdated;
+
+  @Column(nullable = false, columnDefinition = "BOOLEAN DEFAULT false")
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: `nullable = false` is implied

File: src/test/java/uk/gov/ons/census/casesvc/service/CaseServiceTest.java [hunk 6]
```diff
@@ -31,6 +31,7 @@
 import uk.gov.ons.census.casesvc.model.repository.CaseRepository;
 
 @RunWith(MockitoJUnitRunner.class)
+@ActiveProfiles("test")
```
Comment (NickGrantONS) [inline src/test/java/uk/gov/ons/census/casesvc/service/CaseServiceTest.java]: This isn't needed/used anymore

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java (hunk 5 above)
Comment (ryangrundy7) [reply src/main/java/uk/gov/ons/census/casesvc/model/entity/Case.java]: Want me to change the rest of the columns to remove `nullable = false`

**Pull request merged**